packages/dataviews-client/src/resolve-dataviews.ts

```diff
@@ -68,39 +68,141 @@ export const mergeWorkspaceUrlDataviewInstances = (
+    if (dataviewInstance.config?.type !== Generators.Type.Track) return dataviewInstance
+
+    const { track, info, events } = getTrackDataviewDatasetConfigs(dataviewInstance)
+    let preparedDatasetConfigs
+
+    if (trackDatasetConfigsTransform) {
+      preparedDatasetConfigs = trackDatasetConfigsTransform({ track, info, events })
+    } else {
+      preparedDatasetConfigs = [track, info, ...events] as DataviewDatasetConfig[]
+    }
+
+    const preparedDataview = {
+      ...dataviewInstance,
+      datasetsConfig: preparedDatasetConfigs,
+    }
+    return preparedDataview
+  })
+  return preparedDataviewsInstances
+}
+
+/**
+ * Collect available datasetConfigs from dataviews and prepare resource queries
+ */
+export const resolveResourcesFromDatasetConfigs = (
+  dataviews: UrlDataviewInstance[]
+): Resource[] => {
+  return dataviews
+    .filter((dataview) => dataview.config?.type === Generators.Type.Track)
+    .flatMap((dataview) => {
+      if (!dataview.datasetsConfig) return []
+      return dataview.datasetsConfig.flatMap((datasetConfig) => {
+        const dataset = dataview.datasets?.find((dataset) => dataset.id === datasetConfig.datasetId)
+        if (!dataset) return []
+        const url = resolveEndpoint(dataset, datasetConfig)
+        if (!url) return []
+        return [{ dataset, datasetConfig, url, dataviewId: dataview.dataviewId as number }]
+      })
+    })
+}
+
+export const resolveDataviewDatasetResources = (
+  dataview: UrlDataviewInstance,
+  datasetTypeOrId: DatasetTypes | DatasetTypes[] | string
+): Resource[] => {
+  const isArray = Array.isArray(datasetTypeOrId)
+  const isType = isArray || Object.values(DatasetTypes).includes(datasetTypeOrId as DatasetTypes)
   let types: DatasetTypes[]
   if (isType) {
-    types = isArray ? (typeOrId as DatasetTypes[]) : [typeOrId as DatasetTypes]
+    types = isArray ? (datasetTypeOrId as DatasetTypes[]) : [datasetTypeOrId as DatasetTypes]
   }
 
-  const dataset = isType
-    ? dataview.datasets?.find((dataset) => types.includes(dataset.type))
-    : dataview.datasets?.find((dataset) => dataset.id === typeOrId)
+  const dataviewDatasets = dataview.datasets || []
 
-  if (!dataset) return {}
-  const datasetConfig = dataview?.datasetsConfig?.find(
-    (datasetConfig) => datasetConfig.datasetId === dataset.id
-  )
+  const datasets = isType
+    ? dataviewDatasets.filter((dataset) => types.includes(dataset.type))
+    : dataviewDatasets.filter((dataset) => dataset.id === datasetTypeOrId)
 
-  if (!datasetConfig) return {}
-
-  const url = resolveEndpoint(dataset, datasetConfig)
+  const resources = datasets?.flatMap((dataset) => {
+    const datasetConfig = dataview?.datasetsConfig?.find(
+      (datasetConfig) => datasetConfig.datasetId === dataset.id
+    )
+    if (!datasetConfig) return []
+    const url = resolveEndpoint(dataset, datasetConfig)
+    if (!url) return []
+    return [{ dataset, datasetConfig, url, dataviewId: dataview.dataviewId as number }]
```

Comment:
I like having the dataviewId associated in the resource, this will simplify A LOT reading resources.
Thanks!